### PR TITLE
feat: Filter Arguments with $filters in Config\Filters

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add Filter Arguments with `$filters` in `Config\Filters`
  - the route filter's arguments take precedence.
  - If the arguments are already defined, throws an exception.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
If both the Route filter and Config filter are applied, should the Route filter's arguments take precedence?

Or, when multiple Config filters are applied, only one is actually executed, what is the arguments in that case?

File: system/Filters/Filters.php [hunk 1]
```diff
@@ -463,20 +493,59 @@ protected function processFilters(?string $uri = null)
...
+     * @param string $name      filter alias
+     * @param array  $arguments filter arguments
+     * @param bool   $check     if true, check if already defined
+     */
+    private function registerArguments(string $name, array $arguments, bool $check = true): void
+    {
+        if ($arguments !== []) {
+            if ($check && array_key_exists($name, $this->arguments)) {
+                throw new ConfigException(
+                    '"' . $name . '" has already arguments: '
```
Comment (MGatner) [inline system/Filters/Filters.php:534]:
```suggestion
                    '"' . $name . '" already has arguments: '
```

File: system/Filters/Filters.php [hunk 2]
```diff
@@ -463,20 +493,59 @@ protected function processFilters(?string $uri = null)
...
+                    . (($this->arguments[$name] === null) ? 'null' : implode(',', $this->arguments[$name]))
```
Comment (MGatner) [inline system/Filters/Filters.php:535]: How can this get a `null` argument?

**Review by MGatner — CHANGES_REQUESTED** [review]
Awesome! Long-requested feature.

File: system/Filters/Filters.php (hunk 2 above)
Comment (kenjis) [reply system/Filters/Filters.php:535]: 1. Set route filter with no argument.
2. Set config filter with arguments.

Added a test case: https://github.com/codeigniter4/CodeIgniter4/pull/7159/files#diff-b7622ae6db0403f01050df526b06419d30ca51b85ad2220da78fec16aaa9f3c8R285

**Review by datamweb — APPROVED** [review]
Very good.
I like this PR 💯 .

**Review by MGatner — APPROVED** [review]
LGTM!

**Pull request merged**